esque/ruleparser/helpers.py

```diff
@@ -0,0 +1,119 @@
+import datetime
+
+
+def zero_or_more(element):
+    return "(" + element + ")*"
+
+
+def one_or_more(element):
+    return "(" + element + ")+"
+
+
+def zero_or_one(element):
+    return "(" + element + ")?"
+
+
+def exactly(element, count):
+    return "(" + element + "){" + str(count) + "}"
+
+
+def either_of(iterable):
+    return "((" + ")|(".join(iterable) + "))"
+
+
+def is_float(input_string: str):
+    try:
+        float(input_string)
+        return True
+    except ValueError:
+        return False
+
+
+def is_int(input_string: str):
+    try:
+        int(input_string)
+        return True
+    except ValueError:
+        return False
+
+
+def is_any_numeric_type(input_string: str):
+    return is_float(input_string) or is_int(input_string)
+
+
+def is_date_string(input_string: str):
+    try:
+        datetime.datetime.strptime(input_string, "%Y-%m-%d")
+        return True
+    except ValueError:
+        return False
+
+
+def is_date_time_string(input_string: str):
+    try:
+        datetime.datetime.strptime(input_string, "%Y-%m-%dT%H%M%S")
+        return True
+    except ValueError:
+        return False
+
+
+def is_boolean(input_string: str):
+    return to_boolean(input_string) is not None
+
+
+def to_boolean(input_string: str):
+    """Returns True, False or None, based on the contents of the input_string (any form of capitalization is allowed)"""
+    if input_string.lower() == "true":
+        return True
+    elif input_string.lower() == "false":
+        return False
+    else:
+        return None
+
+
+def to_date_time(input_string: str):
+    """This method will try to convert the string into a datetime object even if it is shorter than necessary.
```

Comment:
What about using dateutil.parser.parse. I think that function does the same thing

Reply:
then we don't have to test it :)

Reply:
Or use pendulum. it’s already a requirement anyway.

Reply:
Pendulum already has the same functionality, so this issue resolved itself :)